nano/test_common/network.hpp

```diff
@@ -15,4 +22,114 @@ namespace transport
 
 /** Waits until a TCP connection is established and returns the TCP channel on success*/
 std::shared_ptr<nano::transport::channel_tcp> establish_tcp (nano::system &, nano::node &, nano::endpoint const &);
+
+struct simple_socket : public boost::enable_shared_from_this<simple_socket>
+{
```

Comment:
Should this class in its own file?

Reply:
moved it back to the test network.tcp_connection as it is only used in there. These classes were only intended to organize the code that was written there before.